public/locations-list-map.js

```diff
@@ -770,34 +787,37 @@ function centerMapToMarkersNearCoords(latitude, longitude) {
       distance = distance + 1;
     }
 
-    markerDistances[distance] = marker;
-
-    distances.push(distance);
+    markerDistances.set(distance, marker);
   }
 
-  // sort the distances and set bounds to closest three
-  distances.sort((a, b) => a - b);
+  // order markerDistances by key (distance)
+  let distances = [...markerDistances.keys()].sort((a,b) => a -b);
+  // return array of markers in order of distance ascending
+  return distances.map((distance) => markerDistances.get(distance));
```

Comment:
If getMarkersByDistanceFrom takes the desired length as a param, you can slice the top n off the array and just return those markers instead of all of them.